databricks/koalas/base.py

```diff
@@ -944,6 +944,9 @@ def value_counts(self, normalize=False, sort=True, ascending=False, bins=None, d
         if bins is not None:
             raise NotImplementedError("value_counts currently does not support bins")
 
+        # if given Series is a subset of DataFrame, we should've specified a proper subset.
+        if self._kdf._internal is not self._internal:
+            self = _col(self.to_frame())
```

Comment:
This doesn't fix all the problems.

```py
>>> kser = ks.Series([1,2,3], index=[2,None,5])
>>> kser
2.0    1
NaN    2
5.0    3
Name: 0, dtype: int64
>>> kser.value_counts()
1    1
3    1
Name: 0, dtype: int64
```

whereas:

```py
>>> kser.to_pandas().value_counts()
3    1
2    1
1    1
Name: 0, dtype: int64
```

Reply:
@ueshin thanks for the review. fixed & added tests cases :)